### PR TITLE
Fix deprecations when running against ODM 1.3

### PR DESCRIPTION
**Pull request by alcaeus**
Apparently, some deprecations slipped through the cracks.

* Indirect notices are skipped - we generally can't fix these (think vendor code triggering deprecation notices)
* A single usage of a deprecated document repository class is ignored as fixing it requires bumping the minimum ODM version (which can't be done in a patch release)
* All other deprecations were fixed in the test suite.

### CODE REVIEW

File: Tests/Mapping/Driver/YamlDriverTest.php [hunk 1]
```diff
@@ -5,6 +5,9 @@
 
 use Doctrine\Bundle\MongoDBBundle\Mapping\Driver\YamlDriver;
 
+/**
+ * @group legacy
```
Comment (malarzm) [inline Tests/Mapping/Driver/YamlDriverTest.php]: this seems unrelated, is it some magic group name that allows deprecations to be present?

Comment (alcaeus) [reply Tests/Mapping/Driver/YamlDriverTest.php]: The group tells the Symfony deprecation handler to ignore all deprecation notices that come up during the execution of this test. This can be done to specifically test legacy behaviour without having to define an exception for the deprecation in configuration.
